packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -642,37 +644,45 @@ function expandParameterPlaceholders(
 ): string {
   const azureSolutionSettings = ctx.projectSettings?.solutionSettings as AzureSolutionSettings;
   const plugins = getActivatedResourcePlugins(azureSolutionSettings); // This function ensures return result won't be empty
-  const availableVariables: Record<string, string> = {};
+  const profileVariables: Record<string, Record<string, any>> = {};
+  const availableVariables: Record<string, Record<string, any>> = {};
   // Add plugin contexts to available variables
   for (const plugin of plugins) {
     const pluginContext = getPluginContext(ctx, plugin.name);
+    const pluginVariables: Record<string, string> = {};
     for (const configItem of pluginContext.config) {
       if (typeof configItem[1] === "string") {
         // Currently we only config with string type
-        const variableName = `${normalizeToEnvName(plugin.name)}__${normalizeToEnvName(
-          configItem[0]
-        )}`;
-        availableVariables[variableName] = configItem[1];
+        pluginVariables[configItem[0]] = configItem[1];
       }
     }
+    profileVariables[plugin.name] = pluginVariables;
   }
+  if (expandSecrets === false) {
+    escapeSecretPlaceholders(profileVariables);
+  }
+  availableVariables[profileName] = profileVariables;
   // Add solution config to available variables
   const solutionConfig = ctx.envInfo.profile.get(GLOBAL_CONFIG);
   if (solutionConfig) {
+    const solutionVariables: Record<string, string> = {};
     for (const configItem of solutionConfig) {
       if (typeof configItem[1] === "string") {
         // Currently we only config with string type
-        const variableName = `SOLUTION__${normalizeToEnvName(configItem[0])}`;
-        availableVariables[variableName] = configItem[1];
+        solutionVariables[configItem[0]] = configItem[1];
       }
     }
+    availableVariables[solutionName] = solutionVariables;
```

Comment:
should be availableVariables[profileName], because "solution" is also part of profile.dev.json